charts/jaeger/templates/ingester-deploy.yaml

```diff
@@ -47,64 +47,19 @@ spec:
 {{ toYaml .Values.ingester.securityContext | indent 10 }}
         image: {{ .Values.ingester.image }}:{{ .Values.tag }}
         imagePullPolicy: {{ .Values.ingester.pullPolicy }}
-        env:
+        args:
           {{- range $key, $value := .Values.ingester.cmdlineParams }}
-          - name: {{ $key | replace "." "_" | replace "-" "_" | upper | quote }}
-            value: {{ $value | quote }}
+          - --{{ $key }}={{ $value }}
```

Comment:
same as above

Reply:
same as above :p https://www.jaegertracing.io/docs/1.17/cli/#jaeger-ingester

FWIW even --help seems to be able to take true/false

Reply:
But think about it when you run the executable you only ever provide `--help` and it changes the value from `false` to `true`. It makes sense for all boolean flags to behave this way

Reply:
So we have two choices:

* have users explicitly set true for the boolean flags they wish to enable
or
* templating trickery to detect if a value was provided with the key only for boolean flags that are disabled by default

Reply:
tbh I'm not sure how to achieve the latter. Any suggestions?

Reply:
The easiest possible option that comes to my mind is treating the entire flag as one string. Then it's just a string iteration:

```
--some-bool-flag
--some-int-flag=5
```

i.e. don't split by key/value.

Reply:
or maybe conditionally print `={{ $value }}` if the value is not nil.

Reply:
Found an example frmo a chart in the stable repo:

```
          {{- if $value }}
            - --{{ $key }}={{ $value }}
          {{- else }}
            - --{{ $key }}
          {{- end }}
```

Will try that!

Reply:
The above works with the caveat that the key only value must be defined with a trailing `:` as such:

```
    cmdlineParams:
      foo:
      bar: baz
```

which renders:

```
          - --foo
          - --bar=baz
```

Good enough? 😬

Reply:
Gonna suppose so 😅 here's https://github.com/jaegertracing/helm-charts/pull/75/commits/8d36352e78b758e2fd7eb4d3006713a16a4a8366

Reply:
https://github.com/jaegertracing/helm-charts/pull/75/commits/64352aea620b02d8cb8c13ae786df842af8348c3

Voilà!